src/lib/PickerComponent.android.js

```diff
@@ -34,6 +34,20 @@ export class PickerComponent extends React.Component {
     if (this.props.onValueChange) this.props.onValueChange(value);
   };
 
+  componentDidUpdate = () => {
+    const value = this.props.value;
+    const values = normalisePicklistValue(value);
+    let optionValues = this.props.options.map(o => o.value || o.constant);
+    const validValues = values.filter(v => optionValues.indexOf(v) !== -1);
```

Comment:
We don't need `validValues` until we detect invalid values

Reply:
(I know there's already loads of duplication but) isn't there a higher level helper function here to be a bit more DRY? Along the lines of `sanitisePicklistValue(value, options)` that could return a tuple `[valid: boolean, sanitised: string]`

Reply:
^ I think this would make  a bit clearer what's going on, and could have test cases around it